src/tracer/MOM_neutral_diffusion.F90

```diff
@@ -266,44 +279,107 @@ subroutine neutral_diffusion_calc_coeffs(G, GV, h, T, S, EOS, CS)
   real, dimension(SZI_(G),SZJ_(G),SZK_(G)+1) :: Tint ! Interface T (degC)
   real, dimension(SZI_(G),SZJ_(G),SZK_(G)+1) :: Sint ! Interface S (ppt)
   real, dimension(SZI_(G),SZJ_(G),SZK_(G)+1) :: Pint ! Interface pressure (Pa)
-  real, dimension(SZI_(G),SZJ_(G),SZK_(G)+1) :: dRdT ! Interface thermal expansion coefficient (kg/m3/degC)
-  real, dimension(SZI_(G),SZJ_(G),SZK_(G)+1) :: dRdS ! Interface haline expansion coefficient (kg/m3/ppt)
+  ! Allocatable variables for discontinuous reconstructions
+  ! Note here that the underscript 't' for top interface is equivalent to 'left' edge in the PPM literature
+  real, dimension(:,:,:,:), allocatable        :: T_i     ! Top edge reconstruction of temperature (degC)
+  real, dimension(:,:,:,:), allocatable        :: S_i     ! Top edge reconstruction of salinity (ppt)
+  real, dimension(:,:,:,:), allocatable        :: dRdT_i     ! dRho/dT (kg/m3/degC) at top edge
+  real, dimension(:,:,:,:), allocatable        :: dRdS_i     ! dRho/dS (kg/m3/ppt) at top edge
+  real, dimension(:,:),     allocatable        :: ppoly_coefficients ! PPM coefficients
+  ! Allcoatable variables for continuous reconstructions
+  real, dimension(:,:,:), allocatable        :: dRdT ! dRho/dT (kg/m3/degC) at interface
+  real, dimension(:,:,:), allocatable        :: dRdS ! dRho/dS (kg/m3/ppt) at interfaces
+
+  if (CS%continuous_reconstruction) then
+    if (.not. ALLOCATED(dRdT)) ALLOCATE(dRdT(SZI_(G),SZJ_(G),SZK_(G)+1))
```

Comment:
Whoops, this was an oversight before making the PR. I think when I wrote it I was trying to be super-safe that there was no memory leak.